### PR TITLE
Image optimisation

### PR DESCRIPTION
**Pull request by franz899**
For jpeg, png and svg files

### CODE REVIEW

File: gulp/images.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const size = require("gulp-size");
+const imagemin = require("gulp-imagemin");
+const imageminMozjpeg = require("imagemin-mozjpeg");
+const imageminPngquant = require("imagemin-pngquant");
+const imageminSvgo = require("imagemin-svgo");
+
+// config
+const paths = require("../config.json").paths;
+
+function imageOptim() {
```
[the remark this answers is not in the document]
Comment (franz899) [reply gulp/images.js]: The reason was that `static/assets` contains both images and other types. Is a step more, but is going to copy across everything at once.

File: gulp/images.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// If in PRODUCTION perform some magic
+function images(done) {
+  return gulp
+    .src(paths.images.src, { dot: true })
+    .pipe(gulp.dest(paths.images.dest))
+    .pipe(size());
+}
+
+module.exports = {
+  imageOptim,
```
[the remark this answers is not in the document]
Comment (franz899) [reply gulp/images.js]: I actually chose to cherry pick them so the eventual internal one are not going to be exposed 😅

File: package.json [hunk 3]
```diff
@@ -11,7 +11,8 @@
     "test-watch": "jest --coverage --watch",
     "cypress:open": "cypress open",
     "cypress:cli": "cypress run",
-    "cypress:travis": "http-server dist -c-1 -s & cypress run"
+    "cypress:travis": "http-server dist -c-1 -s & cypress run",
+    "image-optim": "gulp image-optim"
```
[the remark this answers is not in the document]
Comment (franz899) [reply package.json]: Nope, I've added it just for debugging purposes

File: gulp/images.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        imageminPngquant({ quality: "65-80" }),
+        imageminSvgo({ plugins: [{ removeViewBox: false }] })
+      ])
+    )
+    .pipe(gulp.dest(paths.uncompressed.dest));
+}
+
+// Copy all images
+// If in PRODUCTION perform some magic
+function images(done) {
```
[the remark this answers is not in the document]
Comment (franz899) [reply gulp/images.js]: That was just copy/paste from gulp.js

I'll remove it.

**Comment by franz899** [comment]
The point of moving the images in another folder is that there is already a task that copies everything that is inside `static` into `dist`. That task cannot live together with the image optimisation one.

We need a folder where to drop all the images that we want to use that will respects the folder structure of `dist`.
The images not optimised images cannot live inside `static`.

scripts is such a common name that I thought it was a file used to put generic functions. I can move `imageOptim` again.

I used `imageOptim` because there is an app called `ImageOptim` that is quite famous and it seemed like a straightforward way of explaining what it does.

**Comment by franz899** [comment]
Ok, now everything is again into static. This solution looks neat but is not as easy to read.

File: gulpfile.js [hunk 5]
```diff
@@ -75,13 +70,29 @@ function clean(done) {
...
+
 function watch(done) {
   gulp.watch(
     paths.scripts.modules,
     gulp.series(scripts.copyModules, scripts.bundle, browserSync.reload)
   );
   gulp.watch(paths.styles.src, gulp.series(styles, browserSync.reloadCSS));
-  gulp.watch(paths.images.src, gulp.series(images, browserSync.reload));
+  gulp.watch(
+    paths.uncompressed.src,
```
Comment (ThePeach) [inline gulpfile.js]: I think this is something of the past.

**Pull request merged**